parsigs/parse_sig_api.py

```diff
@@ -92,14 +92,12 @@ def _flatmap(func, iterable):
 
 
 def _parse_sigs(sig_lst, model: Language):
-    return _flatmap(lambda sig: _parse_sig(sig, model), sig_lst)
-
+    preprocessed_sigs = [_pre_process(sig) for sig in sig_lst]
+    model_outputs = [doc for doc in model.pipe(preprocessed_sigs)]
```

Comment:
@julian-pani did you test it for improvement?

Reply:
@royashcenazi to be honest, I don't remember. It was a crazy week and I tried a bunch of things in parallel to make Atlantic work.

Reply:
@julian-pani current usage is spawning one process, add `-1` to n_processes params to spawn as the number of cpus available
![Screenshot 2024-10-31 at 16 26 50](https://github.com/user-attachments/assets/0472ad79-bd4e-41c1-a2c1-ea14ccb836fb)

```
    def pipe(  # noqa: F811
        self,
        texts: Union[
            Iterable[Union[str, Doc]], Iterable[Tuple[Union[str, Doc], _AnyContext]]
        ],
        *,
        as_tuples: bool = False,
        batch_size: Optional[int] = None,
        disable: Iterable[str] = SimpleFrozenList(),
        component_cfg: Optional[Dict[str, Dict[str, Any]]] = None,
        n_process: int = 1,
```

Reply:
Also, it seems it is not always recommnded to spawn new process (for example when working with GPU) 
So seems like it is best to create a flag so that user can choose if they want to multiprocess sigs parsing